classes/file_manager.php

```diff
@@ -0,0 +1,186 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Class to manage seb files, used in part to abstract some of the Moodle file manager logic.
+ *
+ * @package    quizaccess_seb
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+use stored_file;
+
+defined('MOODLE_INTERNAL') || die();
+
+class file_manager {
+
+    /** Component the file will be stored in.  */
+    const COMPONENT = 'mod_quiz';
+
+    /** File area in the module to store files.  */
+    const FILE_AREA = 'quizaccess_seb_quizsettings';
+
+    /**
+     * Get a stored file object representing a draft file saved from a form.
+     *
+     * @param string $itemid
+     * @return stored_file|null
+     * @throws \coding_exception
+     */
+    public function get_form_file_by_itemid(string $itemid) {
+        global $USER; // Uploaded files initially have a user context.
```

Comment:
maybe move this comment to $context = \context_user::instance($USER->id); and explain why.

Reply:
Done.